### PR TITLE
157292: Fixing cache expiration check

### PR DESCRIPTION
**Pull request by dachew**
Fixing the cache expiration check, and adding some unit tests.

Reviewers:
- [x] @SQLBob 
- [x] @gstamm-ot 
- [x] @creg

### CODE REVIEW

File: Druq.UnitTests/Framework/Database/CachingDatabaseFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                .Verify
+                (
+                    m => m.GetPartitionDb(partition, false),
+                    Times.Once(),
+                    "The item should not be expired if it was just created"
+                );
+        }
+
+        [Test]
+        public void Test_GetParitionDb_subsequent_calls_are_correct()
```
Comment (gstamm-ot) [inline Druq.UnitTests/Framework/Database/CachingDatabaseFactoryTests.cs]: It should be "Partition", not "Parition"

Comment (dachew) [reply Druq.UnitTests/Framework/Database/CachingDatabaseFactoryTests.cs]: Dang it.  Yes, it should be!  Thanks.

**Comment by gstamm-ot** [comment]
:shipit:

**Comment by SQLBob** [comment]
🚀

File: Druq.Framework/Database/CachingDatabaseFactory.cs [hunk 2]
```diff
@@ -24,8 +24,8 @@ public IHubDb GetHubDb(bool usePersistentConnection = false)
         public IPartitionDb GetPartitionDb(Partition partition, bool usePersistentConnection = false)
         {
             //refresh every 5 minutes to get name changes
-            var cacheItem = _partitionDbs.GetOrAdd(partition, p => new CacheItem(_backingDatabaseFactory.GetPartitionDb(p)));
-            if (cacheItem.ExpiresOn > Time.Now)
+            var cacheItem = _partitionDbs.GetOrAdd(partition, p => new CacheItem(_backingDatabaseFactory.GetPartitionDb(p, usePersistentConnection)));
+            if (cacheItem.ExpiresOn < Time.Now)
```
Comment (creg) [inline Druq.Framework/Database/CachingDatabaseFactory.cs]: This should be using Bcl.Time.Now (and probably ought to be using UtcNow). If we switch to Bcl, then we can write a test confirming the Cache refreshes on expiration.

Comment (dachew) [reply Druq.Framework/Database/CachingDatabaseFactory.cs]: I don't disagree.  However, in classic Drew fashion, that's not the .Net time object.  It's this: https://github.com/daptiv/Daptiv.Bus/blob/master/Druq.Framework/Time.cs.  And then this is used to set the delegate: https://github.com/daptiv/Daptiv.Bus/blob/master/Druq.App/Program.cs#L36

Comment (creg) [reply Druq.Framework/Database/CachingDatabaseFactory.cs]: Okaay. Well, then NowDelegate can be used to write the test then. Just set the delegate to return different fixed times at different points in the test to simulate the passing of time.

Comment (dachew) [reply Druq.Framework/Database/CachingDatabaseFactory.cs]: done

Comment (creg) [reply Druq.Framework/Database/CachingDatabaseFactory.cs]: :+1:

**Comment by creg** [comment]
:shipit:

**Pull request merged**